gslib/utils/copy_helper.py

```diff
@@ -3711,11 +3736,12 @@ def PerformCopy(logger,
         raise CommandException(message)
     if src_url.IsStream() or src_url.IsFifo():
       src_obj_size = None
-    elif src_obj_metadata and src_obj_metadata.size:
+    elif src_obj_metadata and src_obj_metadata.size and not use_stet:
       # Iterator retrieved the file's size, no need to stat it again.
+      # Unless STET changed the file size.
       src_obj_size = src_obj_metadata.size
     else:
-      src_obj_size = os.path.getsize(src_url.object_name)
+      src_obj_size = os.path.getsize(source_stream_url.object_name)
```

Comment:
Is the behavior verified for rsync? given that rsync does size comparison.

Also, we are not updating the size for downloads, does that have implications with rsync?

Reply:
The use_stet flag wouldn't be accessible with rsync, so how would the new code be triggered?

Reply:
Ah, I though we are supporting it for rsync as well. Is there an expectation that it should work with rsync in the future? maybe we should confirm that requirement from the stet team.

Reply:
I don't think that makes sense fundamentally. To run rsync's file comparison algorithms, we would have to encrypt all the local files before comparing to the cloud. So we would have to operate on all the files before deciding if they should be copied.